synfig-studio/CMakeLists.txt

```diff
@@ -20,31 +20,43 @@ add_subdirectory(ui)
 file(MAKE_DIRECTORY "${SYNFIG_BUILD_ROOT}/share/appdata")
 file(MAKE_DIRECTORY "${SYNFIG_BUILD_ROOT}/share/applications")
 
-STUDIO_INTLTOOL_MERGE(
-    TARGET_NAME appdata_xml
-    INPUT_FILE ${CMAKE_CURRENT_SOURCE_DIR}/org.synfig.SynfigStudio.appdata.xml.in
-    OUTPUT_FILE "org.synfig.SynfigStudio.appdata.xml"
-    INSTALL_DESTINATION "share/appdata"
-)
-
-STUDIO_INTLTOOL_MERGE(
-    DESKTOP
-    TARGET_NAME desktop_file
-    INPUT_FILE ${CMAKE_CURRENT_SOURCE_DIR}/org.synfig.SynfigStudio.desktop.in
-    OUTPUT_FILE "org.synfig.SynfigStudio.desktop"
-    INSTALL_DESTINATION "share/applications"
-)
-
-add_custom_command(
-    TARGET desktop_file POST_BUILD
-    COMMAND sed -i -E \"s|^Icon=\(.*\)|Icon=/usr/share/icons/hicolor/scalable/apps/org.synfig.SynfigStudio.svg|g\"
-        ${SYNFIG_BUILD_ROOT}/share/applications/org.synfig.SynfigStudio.desktop
-)
+# This doesn't work on Windows MSYS builds.
+if(LINUX)
```

Comment:
If I understand correctly, this does not work because `INTLTOOL MERGE EXECUTABLE` is not available.
So it's better to check if `INTLTOOL MERGE EXECUTABLE` is available.

Reply:
As shown in the Appveyor build, it is available [here](https://ci.appveyor.com/project/Synfig/synfig/builds/34171306?fullLog=true#L617). In MSYS, when we use that with `add_custom_command`, it uses `cmd` to run it and since `intltool-merge` is not an executable file, it doesn't recognize it as an internal command. It's how I mentioned it before [here](https://github.com/synfig/synfig/pull/1491#discussion_r453282905).
```
FAILED: output/share/synfig/plugins/add-skeleton-simple/plugin.xml
cmd.exe /C "cd /D D:\msys64\home\user\synfig\cmake-build-msys\synfig-studio\plugins\add-skeleton-simple && D:\msys64\mingw64\bin\intltool-merge -x -u -c D:/msys64/home/user/synfig/cmake-build-msys/synfig-studio/po/.intltool-merge-cache D:/msys64/home/user/synfig/synfig-studio/po D:/msys64/home/user/synfig/synfig-studio/plugins/add-skeleton-simple/plugin.xml.in D:/msys64/home/user/synfig/cmake-build-msys/output/share/synfig/plugins/add-skeleton-simple/plugin.xml"
'D:\msys64\mingw64\bin\intltool-merge' is not recognized as an internal or external command,```

Reply:
> it uses cmd to run it and since intltool-merge is not an executable file
Yes, that is why you are checking if the binary is available or not. Maybe you can do a minimal reproducible example?

Reply:
I made a small repository ([457239709](https://github.com/AYESDIE/457239709)) to reproduce that same issue.

Reply:
Ok, thank you! Now I see the problem.

Reply:
This is example, how it can be fixed:
```
find_program(INTLTOOL_MERGE intltool-merge)

# When MinGW is used without MSYS environment (for example, Ninja
# generator uses the standard Windows `cmd` shell for building), `intltool-merge`
# cannot be started, because it is just a Perl script, and `cmd` does not know how
# to run it. In this case, we need to explicitly add the interpreter to the command.
if(INTLTOOL_MERGE AND MINGW AND NOT MSYS)
  message(STATUS "Fixing intltool-merge command...")
  set(INTLTOOL_MERGE perl ${INTLTOOL_MERGE})
endif()
```